components/action-mgt/org.wso2.carbon.identity.action.execution/src/main/java/org/wso2/carbon/identity/action/execution/ActionExecutorServiceImpl.java

```diff
@@ -92,7 +92,7 @@ public ActionExecutionStatus execute(ActionType actionType, Map<String, Object>
                     .orElse(new ActionExecutionStatus(ActionExecutionStatus.Status.FAILED, eventContext));
         } catch (ActionExecutionRuntimeException e) {
             // todo: add to diagnostics
-            LOG.error("Skip executing actions for action type: " + actionType.name() + ". Error: " + e.getMessage(), e);
+            LOG.debug("Skip executing actions for action type: " + actionType.name(), e);
```

Comment:
Lets add the `if (LOG.isDebugEnabled())` check as well.

Reply:
Well it's not needed for simple string concatenations. Real need of the check is when we are using string formatter as it creates StringBuilder objects

Related discussion: https://stackoverflow.com/questions/963492/in-log4j-does-checking-isdebugenabled-before-logging-improve-performance